core/src/main/java/de/danoeh/antennapod/core/feed/IntraFeedSortOrder.java

```diff
@@ -0,0 +1,65 @@
+package de.danoeh.antennapod.core.feed;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.Comparator;
+
+import de.danoeh.antennapod.core.util.comparator.FeedItemDurationComparator;
+import de.danoeh.antennapod.core.util.comparator.FeedItemPubdateComparator;
+import de.danoeh.antennapod.core.util.comparator.FeedItemTitleComparator;
+
+/**
+ * Provides sort orders to sort a list of episodes within a feed.
+ */
+public enum IntraFeedSortOrder {
+    DATE_OLD_NEW(1, FeedItemPubdateComparator.ascending),
+    DATE_NEW_OLD(2, FeedItemPubdateComparator.descending),
+    EPISODE_TITLE_A_Z(3, FeedItemTitleComparator.ascending),
+    EPISODE_TITLE_Z_A(4, FeedItemTitleComparator.descending),
+    DURATION_SHORT_LONG(5, FeedItemDurationComparator.ascending),
+    DURATION_LONG_SHORT(6, FeedItemDurationComparator.descending);
```

Comment:
Sort order refactoring is probably a work item on its own. Right now, including this PR, there are at least 3 types of sorts that share some similarities.
1. Sort on Podcast Screen: the `IntraFeedSortOrder` enum, its associated comparators, SortDialog UI, etc.
2. Sort on Queue: `util.SortOrder` enum. Conceptually a superset of Podcast Screen sorts. Some of them do not have comparators though (e.g., smart shuffle).
3. Subscription Order (in Nav Drawer, but conceivably podcasts can be sorted on Subscription screen too): some of them, e.g., sorts by pubDate and sorts by name, are analogous to episode sorts above as well.

Reply:
Option 3 is quite different, I think. We can not use the same comparators there because the base object is different. That leaves us with 2 locations only. I think for now, it would be better to just use the existing sort enum (extend with comparators etc). I think this is a better temporary solution than having duplicate enums.

Reply:
To clarify what you mean, it'd be something like the following in the existing `core.util.SortOrder`
```java
public enum SortOrder {
    EPISODE_TITLE_A_Z(1, FeedItemTitleComparator.ASCENDING),
    ...
    DATE_OLD_NEW(3, FeedItemPubDateComparator.ASCENDING),
    ...
    RANDOM(9, null),
    SMART_SHUFFLE_OLD_NEW(10, null)
    ...
```

- the comparator is then `null` for some enum values: smart shuffle, random: not applicable

Reply:
Having null comparators is indeed a bit strange... Still, I think it's better than code duplication.

Reply:
I just realize we could reuse existing logic in queue fragment sort: 

https://github.com/AntennaPod/AntennaPod/blob/fe9f0c8e7e65ba66590780d2344658112692c816/core/src/main/java/de/danoeh/antennapod/core/util/QueueSorter.java#L38

The returned `Permutor` lets caller sort a FeedItem list.

Reply:
Good find. That sounds like even less code duplication :+1: I would then prefer to rename the class to be no longer queue specific.